tests/race_conditions/utils_for_race_tests.py

```
@@ -0,0 +1,332 @@
+
+def assert_no_errors(errors: list[BaseException | None]) -> None:
+    """Re-raise the first error captured by ``run_racers``, if any."""
+    for error in errors:
+        if error is not None:
+            raise error
+
+
+async def eventually(
+    predicate: Callable[[], bool],
+    *,
+    timeout: float = 5.0,
```

Ruff (0.15.20)

[warning] Async function definition with a `timeout` parameter

(ASYNC109)